### PR TITLE
Ap 3296 new proceeding types endpoint

### PR DESCRIPTION
**Pull request by kmahern**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3296)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by colinbruce — COMMENTED** [review]
This is looking really good :+1:
I have left a couple of minor non-blocking nitpicks, but, can you confirm that rswag is working as expected... The v5 documents _only_ have the new proceeding_type endpoint.
Is that what we'd expect to see?

File: spec/requests/assessments_controller_spec.rb [hunk 1]
```diff
@@ -109,7 +109,7 @@
       let(:headers) do
         {
           "CONTENT_TYPE" => "application/json",
-          "Accept" => "application/json; version=5",
+          "Accept" => "application/json; version=6",
```
[the remark this answers is not in the document]
Comment (kmahern) [reply spec/requests/assessments_controller_spec.rb]: 👍 made requested change

File: public/schemas/proceeding_types.json.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            "enum": <%= ProceedingTypeThreshold.valid_ccms_codes.as_json %>
+          },
+          "client_involvement_type": {
+            "enum": <%= CFEConstants::VALID_CLIENT_INVOLVEMENT_TYPES.as_json %>
+          }
+        }
+      }
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (kmahern) [reply public/schemas/proceeding_types.json.erb]: 👍 removed empty line

**Comment by kmahern** [comment]
> The v5 documents _only_ have the new proceeding_type endpoint.
> Is that what we'd expect to see?

@colinbruce I think eventually I would expect to see all of the endpoints under v5 (and possibly nothing for v3 and v4), @stephenrichards and I have created ap-3310 to cover the additional work to the assessments_controller to create the v5 endpoint (and ap-3311 to delete the v3 and v4 endpoints) but it is probably worth having a chat to discuss.

**Review by colinbruce — APPROVED** [review]
LGTM :+1:
Thanks for clarifying the documentation issue

**Pull request merged**